src/Control/Selective/Rigid/Freer.hs

```diff
@@ -91,11 +181,15 @@ getPure = runSelect (const Nothing)
 getEffects :: Functor f => Select f a -> [f ()]
 getEffects = foldSelect (pure . void)
 
+-- | Like 'getEffects' but not requiring @'Functor' f@.
+getEffectsSome :: Select f a -> [Some f]
```

Comment:
It's very tempting to name this `getSomeEffects`, but it would be quite misleading :)

Reply:
Indeed :)